js/ui/map.js

```diff
@@ -197,23 +197,11 @@ var Map = module.exports = function(options) {
         this.style.update(this._classes, {transition: false});
     });
 
-    this.on('style.change', function() {
+    this.on('data', function() {
         this._update(true);
```

Comment:
Not every `data` event should trigger `_update(true)` -- that triggers updating and recalculating, which is necessary only in certain cases.